x/valset-pref/keeper_test.go

```diff
@@ -100,6 +103,63 @@ func (suite *KeeperTestSuite) AllocateRewards(ctx sdk.Context, delegator sdk.Acc
 	suite.Require().NotZero(rewardsAfterAllocation[0].Amount)
 }
 
+func (suite *KeeperTestSuite) SetupLocks(delegator sdk.AccAddress) []lockuptypes.PeriodLock {
+	locks := []lockuptypes.PeriodLock{}
+	// Setup lock
+	coinsToLock := sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 10_000_000)}
+	osmoToLock := sdk.Coins{sdk.NewInt64Coin(appParams.BaseCoinUnit, 10_000_000)}
+	suite.FundAcc(delegator, sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 100_000_000), sdk.NewInt64Coin(appParams.BaseCoinUnit, 100_000_000)})
+
+	// happy lock case
+	TwoWeekDuration, _ := time.ParseDuration("336h")
+	workingLoc, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, delegator, osmoToLock, TwoWeekDuration)
+	suite.Require().NoError(err)
+
+	locks = append(locks, workingLoc)
+
+	// locking with stake denom instad of osmo denom
+	stakeDenomLock, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, delegator, coinsToLock, TwoWeekDuration)
+	suite.Require().NoError(err)
+
+	locks = append(locks, stakeDenomLock)
+
+	// lock case where lock owner != delegation owner
+	suite.FundAcc(sdk.AccAddress([]byte("addr5---------------")), sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 100_000_000)})
+	lockWithDifferentOwner, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, sdk.AccAddress([]byte("addr5---------------")), coinsToLock, TwoWeekDuration)
+	suite.Require().NoError(err)
+
+	locks = append(locks, lockWithDifferentOwner)
+
+	// lock case where the duration != <= 2 weeks
+	MorethanTwoWeekDuration, _ := time.ParseDuration("337h")
+	maxDurationLock, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, delegator, coinsToLock, MorethanTwoWeekDuration)
+	suite.Require().NoError(err)
+
+	locks = append(locks, maxDurationLock)
+
+	// unbonding locks
+	unbondingLocks, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, delegator, coinsToLock, TwoWeekDuration)
+	suite.Require().NoError(err)
+
+	if !unbondingLocks.IsUnlocking() {
+		err := suite.App.LockupKeeper.BeginUnlock(suite.Ctx, unbondingLocks.ID, nil)
+		suite.Require().NoError(err)
+	}
+
+	locks = append(locks, unbondingLocks)
+
+	// synthetic locks
+	syntheticLocks, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, delegator, coinsToLock, TwoWeekDuration)
+	suite.Require().NoError(err)
+
+	err = suite.App.LockupKeeper.CreateSyntheticLockup(suite.Ctx, syntheticLocks.ID, "uosmo", time.Minute, true)
+	suite.Require().NoError(err)
```

Comment:
Im asking because I don't know off the top of my head, does running CreateSyntheticLockup change the lock created via CreateLock into a syntheticLock, or does it make a new lock that is a synthetic lock?

Reply:
Yup it creates a new SynthLock object, using the lockId, lockDuration and bondStatus from the existing lock. Here's the link to the [function](https://github.com/osmosis-labs/osmosis/blob/main/x/lockup/keeper/synthetic_lock.go#L94)

Reply:
I ask because you are calling the lock object in the append. When it was first created it was a normal lock, so I am not sure if calling CreateSyntheticLockup on it actually changes that pointer to a synthetic lock if that makes sense?

Reply:
i see, i believe once syntheticLock is created the lockId points towards the SynthLock object instead of a regular period lock object